target/linux/x86/image/Makefile

```diff
@@ -40,13 +38,17 @@ BOOTOPTS:=$(call qstrip,$(CONFIG_GRUB_BOOTOPTS))
 
 define Build/combined
 	$(CP) $(KDIR)/$(KERNEL_NAME) $@.boot/boot/vmlinuz
-	-$(CP) $(STAGING_DIR_ROOT)/boot/. $@.boot/boot/
-	$(CP) $(STAGING_DIR_IMAGE)/grub2/boot.img $@.boot/boot/grub/
-	$(CP) $(STAGING_DIR_IMAGE)/grub2/$(if $(filter $(1),efi),gpt,$(GRUB2_VARIANT))-core.img \
-		$@.boot/boot/grub/core.img
 	$(if $(filter $(1),efi),
-		$(INSTALL_DIR) $@.boot/efi/boot
-		$(CP) $(STAGING_DIR_IMAGE)/grub2/boot$(if $(CONFIG_x86_64),x64,ia32).efi $@.boot/efi/boot/
+		rm -f $@.grub.img
+		mkfs.fat -n ESP -C $@.grub.img -S 512 1024
+		mmd -i $@.grub.img ::/efi ::/efi/boot
+		mcopy -i $@.grub.img \
+			$(STAGING_DIR_IMAGE)/grub2/boot$(if $(CONFIG_x86_64),x64,ia32).efi \
+			::/efi/boot/boot$(if $(CONFIG_x86_64),x64,ia32).efi,
+		-$(CP) $(STAGING_DIR_ROOT)/boot/. $@.boot/boot/
```

Comment:
Why does this need to be `-`?

I'd rather do a test first to see if the command is viable, and then run it.  I'm suspicious of ignoring command exit statuses...

Reply:
Nothing really, it was beyond me when I first read the original, check line 43. I am pretty sure it can be removed altogether.

Reply:
The entire line makes no sense.

I've just checked and `$(STAGING_DIR_ROOT)/boot/` does not even exist to begin with, hence the `-` prefix ignores the error so I'm dropping the whole line.